lib/elixir/lib/kernel/typespec.ex

```diff
@@ -477,26 +477,26 @@ defmodule Kernel.Typespec do
     typespec({:record, meta, [atom, []]}, vars, caller)
   end
 
-  defp typespec({:record, meta, [atom, fields]}, vars, caller) do
+  defp typespec({:record, meta, [tag, field_specs]}, vars, caller) do
     # We cannot set a function name to avoid tracking
     # as a compile time dependency because for records it actually is one.
-    case Macro.expand({atom, [], [{atom, [], []}]}, caller) do
-      keyword when is_list(keyword) ->
+    case Macro.expand({tag, [], [{}]}, caller) do
```

Comment:
An empty tuple is not a valid AST. Maybe you mean a list with an empty list inside?

Reply:
The idea was to match only in the [last condition](https://github.com/elixir-lang/elixir/blob/master/lib/elixir/lib/record.ex#L323) avoiding to call this method twice (one without parameters and another one like it was before) because there is no return to the function that returns the name and the fields together. Since it's quoted I thought it was safe to do this, but I can replace for two calls.